src/repo.rs

```diff
@@ -58,6 +63,35 @@ extern "C" fn mergehead_foreach_cb(oid: *const raw::git_oid, payload: *mut c_voi
     .unwrap_or(1)
 }
 
+extern "C" fn fetchhead_foreach_cb(
+    ref_name: *const c_char,
+    remote_url: *const c_char,
+    oid: *const raw::git_oid,
+    is_merge: c_uint,
+    payload: *mut c_void,
+) -> c_int {
+    panic::wrap(|| unsafe {
+        let data = &mut *(payload as *mut FetchheadForeachCbData<'_>);
+        let res = {
+            let callback = &mut data.callback;
+
+            let ref_name = str::from_utf8(CStr::from_ptr(ref_name).to_bytes()).unwrap();
+            let remote_url = str::from_utf8(CStr::from_ptr(remote_url).to_bytes()).ok();
```

Comment:
Could this add asserts that the input pointers aren't null?

Reply:
I've added asserts for `ref_name`, `remote_url` and `oid`